### PR TITLE
Add retries to gcloud cli auth configuration

### PR DESCRIPTION
**Pull request by sjahl**

### CODE REVIEW

**Comment by sjahl** [comment]
Closes #2437

File: settings.py [hunk 1]
```diff
@@ -395,7 +396,18 @@
...
                                                capture_output=True, text=True, shell=True).stdout.split('\n')[0]
     if not SERVICE_ACCOUNT_FOR_ANVIL:
-        raise Exception('Error starting seqr - gcloud auth is not properly configured')
+        # attempt to acquire a service account token
+        if os.path.exists('/.config/service-account-key.json'):
+            SERVICE_ACCOUNT_FOR_ANVIL = \
+                re.findall('\[(.*)\]',
+                           subprocess.run(['gcloud', 'auth', 'activate-service-account', '--key-file',
+                                           '/.config/service-account-key.json'],
+                                           capture_output=True, text=True).stderr)[0] # nosec
```
Comment (sjahl) [inline settings.py:405]: i had to add the `# nosec` here to pass codacy. It initially complained about the use of `shell=True`, which i thought was reasonable. But, after doing what I thought it wanted, and using a `shell=False`, it complained about unsanitized command input... but the input here is a static list. Let me know if you have any thoughts on this (or thoughts on how to improve the logic in this section in general, @hanars )

**Review by hanars — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## dev
 
+* Update seqr dockerfile to improve immutability and build automation (REQUIRES IMAGE UPDATE)
+* Adds NPM asset build to the seqr dockerfile
```
Comment (hanars) [inline CHANGELOG.md:6]: I think this was an error and meant as part of a different PR?

**Review by hanars — CHANGES_REQUESTED** [review]

**Comment by sjahl** [comment]
Whoops, yeah, forgot to switch branches!

**Review by hanars — APPROVED** [review]

**Pull request merged**